test/functional/collection_tests.js

```diff
@@ -12,422 +12,405 @@ describe('Collection', function() {
+                  client.close();
+                  done();
+                }
+              );
             });
           });
         });
-      });
-    }
-  });
+      }
+    });
 
-  /**
-   * @ignore
-   */
-  it.skip('should fail to insert due to illegal keys', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
-
-    // The actual test we wish to run
-    test: function(done) {
-      var self = this;
-      var client = self.configuration.newClient(self.configuration.writeConcernMax(), {
-        poolSize: 1
-      });
+    /**
+     * @ignore
+     */
+    it('should perform strict create collection', {
+      metadata: {
+        requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
+      },
+
+      // The actual test we wish to run
+      test: function(done) {
+        client.connect((err, client) => {
+          expect(err).to.not.exist;
+          const db = client.db(this.configuration.db);
+          db.createCollection('test_strict_create_collection', (err, collection) => {
+            test.equal(null, err);
+            test.equal('test_strict_create_collection', collection.collectionName);
+
+            // Creating an existing collection should fail
+            db.createCollection('test_strict_create_collection', { strict: true }, err => {
+              test.ok(err instanceof Error);
+              test.equal(
+                'Collection test_strict_create_collection already exists. Currently in strict mode.',
+                err.message
+              );
 
-      client.connect(function(err, client) {
-        var db = client.db(self.configuration.db);
+              // Switch out of strict mode and try to re-create collection
+              db.createCollection(
+                'test_strict_create_collection',
+                { strict: false },
+                (err, collection) => {
+                  test.equal(null, err);
+                  test.ok(collection.collectionName);
 
-        db.createCollection('test_invalid_key_names', function(err, collection) {
-          // Legal inserts
-          collection.insert(
-            [{ hello: 'world' }, { hello: { hello: 'world' } }],
-            self.configuration.writeConcernMax(),
-            function(err) {
-              test.equal(null, err);
+                  // Let's close the db
+                  client.close();
+                  done();
+                }
+              );
+            });
+          });
+        });
+      }
+    });
 
-              // Illegal insert for key
-              collection.insert({ $hello: 'world' }, self.configuration.writeConcernMax(), function(
-                err
-              ) {
-                test.ok(err instanceof Error);
-                test.equal("key $hello must not start with '$'", err.message);
+    /**
+     * @ignore
+     */
+    it.skip('should fail to insert due to illegal keys', {
```

Comment:
Can we investigate why this test is being skipped? Does it fail?

Reply:
In this test, it has the line test.equal(null, err) which kept failing. After rewriting this line as expect(err).to.not.exist, the test now works. I removed the skip.